app/renderer/components/bookmarks/bookmarksToolbar.js

```diff
@@ -50,83 +49,27 @@ class BookmarksToolbar extends React.Component {
 
   onDrop (e) {
     e.preventDefault()
-    const getClosestFromPos = (clientX, sourceKey) =>
-      dnd.closestFromXOffset(this.bookmarkRefs.filter((bookmarkRef) => {
-        if (!bookmarkRef) {
-          return false
-        }
-        return bookmarkRef.props.bookmarkKey !== sourceKey
-      }), e.clientX)
-    let bookmark = dnd.prepareBookmarkDataFromCompatible(e.dataTransfer)
+    const bookmark = dnd.prepareBookmarkDataFromCompatible(e.dataTransfer)
     if (bookmark) {
-      // Figure out the droppedOn element filtering out the source drag item
-      let bookmarkKey = bookmark.get('key')
-      let tabDrop = false
-
-      // When we have key null is only when we are getting data from TAB transfer type
-      if (bookmarkKey == null) {
-        tabDrop = true
-      }
-
-      const droppedOn = getClosestFromPos(e.clientX, bookmarkKey)
-      if (droppedOn.selectedRef) {
-        const isRightSide = !dnd.isLeftSide(ReactDOM.findDOMNode(droppedOn.selectedRef), e.clientX)
-        const droppedOnKey = droppedOn.selectedRef.props.bookmarkKey
-        const isDestinationParent = droppedOn.selectedRef.state.isFolder && droppedOn && droppedOn.isDroppedOn
-        if (tabDrop) {
-          const parentKey = isDestinationParent ? droppedOnKey : null
-          bookmark = bookmark.set('parentFolderId', parentKey)
-          appActions.addBookmark(bookmark)
-        } else {
-          if (bookmark.get('type') === siteTags.BOOKMARK_FOLDER) {
-            appActions.moveBookmarkFolder(bookmarkKey, droppedOnKey, isRightSide, isDestinationParent)
-          } else {
-            appActions.moveBookmark(bookmarkKey, droppedOnKey, isRightSide, isDestinationParent)
-          }
-        }
-        dnd.onDragEnd()
+      const droppedOn = bookmarkUtil.getClosestFromPos(
+        this.bookmarkRefs,
+        e.clientX,
+        bookmark.get('key')
+      )
+      const currentNode = ReactDOM.findDOMNode(droppedOn.selectedRef)
+      const isRightSide = dnd.isRightSide(currentNode, e.clientX)
+
+      if (droppedOn && droppedOn.selectedRef) {
+        appActions.onDropBookmark(
+          bookmark,
+          droppedOn.selectedRef.props.bookmarkKey,
+          droppedOn.selectedRef.state.isFolder,
+          droppedOn.isDroppedOn,
+          isRightSide
+        )
       }
-      return
+      dnd.onDragEnd()
     }
-
-    const droppedOn = getClosestFromPos(e.clientX, undefined)
```

Comment:
iirc this logic was used to allow links from the webview to be made as bookmarks. Given this doesn't work anymore code was removed. Commit that made this removal is 86c1ac42ff6ef5aa026cbfa3eae5556d44141479 in case we need it back.

per git blame cc @bbondy @NejcZdovc if there's other something I missed